src/sentry/grouping/component.py

```diff
@@ -23,33 +23,38 @@
 }
 
 
-def _calculate_contributes(values: Sequence[str | GroupingComponent]) -> bool:
+def _calculate_contributes(values: Sequence[str | int | BaseGroupingComponent]) -> bool:
     for value in values or ():
-        if not isinstance(value, GroupingComponent) or value.contributes:
+        if not isinstance(value, BaseGroupingComponent) or value.contributes:
             return True
     return False
 
 
-class GroupingComponent:
+class BaseGroupingComponent:
     """A grouping component is a recursive structure that is flattened
     into components to make a hash for grouping purposes.
     """
 
+    id: str = "default"
+    hint: str | None
+    contributes: bool | None
+    values: Sequence[str | int | BaseGroupingComponent]
+
     def __init__(
         self,
-        id: str,
+        id: str | None = None,
```

Comment:
why would we ever pass in id if there are specific subclasses for each grouping component?